proposals/records/3-migration-strategy.md

```diff
@@ -0,0 +1,106 @@
+# Migration Strategy
+
+This is a proposal for how we manage registering and running migrations on the database.
+
+The proposal focuses on providing an automated way to migrate data and index nodes.
+
+
+## Current solution/situation
+
+We currently do not have any cohesive strategy around migrations on NucliaDB.
+It is currently managed through manually running scripts against the cluster.
+
+Additionally, the ingest grpc service has a method `CleanAndUpgradeKnowledgeBoxIndex`
+which only deletes a shard and recreates it. Reindexing data there needs to be
+done manually.
+
+
+### Types of data migrations
+
+- Blob storage
+- K/V Storage
+- IndexNodes
+
+## Proposed Solution
+
+1. Build a framework for registering and running migrations on the platform.
+2. Require all changes to to be b/w compatible
+3. Migration are not reversible(register a new migration if you need to "undo")
+
+
+### Types of migrations supported
+
+- IndexNode shard rollovers: create new shards, index all data to new shards, swap active shards over
+- manual: manually run a snippet of code against every knowledge box
```

Comment:
do you mean "custom" instead (custom snippets of code that tweak/fix data)? 
"manual" to me implies "not automated".